pkg/compiler/compiler.go

```diff
@@ -88,9 +94,47 @@ func (c *Compiler) Compile() error {
 		}
 		c.Document.OutputFiles = append(c.Document.OutputFiles, files...)
 	}
+	err := c.createConfigOutputFile()
+	if err != nil {
+		return errors.Wrap(err, "Unable to output Klotho configuration file")
+	}
 	return c.Document.OutputTo(c.Document.Configuration.OutDir)
 }
 
+func (c *Compiler) createConfigOutputFile() error {
+	c.Document.Configuration.UpdateForResources(c.Document.Constructs.ListConstructs())
+	buf := new(bytes.Buffer)
+	var err error
+	var ext string
```

Comment:
Minor: we don't really need this var, since it's always just `c.Document.Configuration.Format` for any case where we don't return an error.